### PR TITLE
Match Interface_Draw (z_parameter OK)

### PR DESCRIPTION
**Pull request by engineer124**
This was an adventure to solve. Had to throw a bunch of temps everywhere to get save registers to play nicely. Also had to break up two `WREG(90)` so I could assign a temp to part of the integer in the Macro, I'm sure there's a better solution out there but I just left a comment for now. Oh and also throw in a same-line meme. 

A couple small non-equivalent code was found but all were minor. 

 Actually remembered to remove the asm this time :)

edit: apologies in advance, seems like my clang-format isn't working so `./format` didn't work

### CODE REVIEW

**Review by Dragorn421 — APPROVED** [review]
Fantastic job, thanks a lot for matching this monstrosity

File: src/code/z_parameter.c [hunk 1]
```diff
@@ -3749,13 +3751,15 @@ void Interface_Draw(GlobalContext* globalCtx) {
...
                 OVERLAY_DISP =
-                    Gfx_TextureIA8(OVERLAY_DISP, gClockIconTex, 16, 16, ((void)0, gSaveContext.timerX[svar6]),
-                                   ((void)0, gSaveContext.timerY[svar6]) + 2, 16, 16, 1 << 10, 1 << 10);
+                    Gfx_TextureIA8(OVERLAY_DISP, gClockIconTex, 16, 16, svar5,
+                                   svar2 + 2, 16, 16, 1 << 10, 1 << 10);
 
                 // Timer Counter
                 gDPPipeSync(OVERLAY_DISP++);
-                gDPSetCombineMode(OVERLAY_DISP++, G_CC_MODULATEIA_PRIM, G_CC_MODULATEIA_PRIM);
+                gDPSetCombineLERP(OVERLAY_DISP++, 0, 0, 0, PRIMITIVE, TEXEL0, 0, PRIMITIVE, 0, 0, 0, 0, PRIMITIVE, TEXEL0, 0, PRIMITIVE, 0);
```
Comment (Zelllll) [inline src/code/z_parameter.c:3762]: Why was this changed here?

**Review by Zelllll — APPROVED** [review]
Overall while I do approve of this matching, I'm a little concerned that we will have a hard time matching this on retail because some things look a bit fake

File: src/code/z_parameter.c (hunk 1 above)
Comment (engineer124) [reply src/code/z_parameter.c:3762]: This was one of three non-equivalent lines I found. The asm produced by `gDPSetCombineMode` gave the wrong asm, and going back to gfxdis gave me a different macro which gave the correct asm. 

Also I agree there's a little fakery going on. I'm mainly concerned about the `WREG(90)` macro and possibly the same-line meme. Definitely worth searching for a better match in the next round of clean-up

Comment (Zelllll) [reply src/code/z_parameter.c:3762]: Ah ok. Another thing is that you removed ((void)0,X) which is eventually going to be added pretty much everywhere because there is evidence that Nintendo used macros for accessing values in save context but anyways that is something we can find out another time, great job on finally getting this matched

Comment (fig02) [reply src/code/z_parameter.c:3762]: its only needed for specific fields of the save context, and i dont think its commonly needed for the ones it was removed from here. so its probably fine
but indeed, we will find out later

**Review by fig02 — APPROVED** [review]
am also not a fan of the direct access to `gGameInfo->data`, but until we have some better ideas this will have to do. a match is nice to have

**Comment by engineer124** [comment]
> am also not a fan of the direct access to `gGameInfo->data`, but until we have some better ideas this will have to do. a match is nice to have

Should I add a comment or two highlighting this in the code and possibly add a `TODO: ` comment?

**Comment by Zelllll** [comment]
Why is game info accessed directly now?

**Comment by engineer124** [comment]
> Why is game info accessed directly now?

`svar3 = 18`
...
`gGameInfo->data[((svar3 * 6) * 16) + 90] / 10; // WREG(90)`

My really dumb match involved needing to set part of the macro to a save register to get the code to match. Replacing any of the `svar3` with 18 results in a non-match.

**Comment by Zelllll** [comment]
In that case it sorta seems like the reg macros could be broken somehow or something tbh

**Comment by engineer124** [comment]
> In that case it sorta seems like the reg macros could be broken somehow or something tbh

It definitely feels like I haven't explored this part of the code enough after getting a match, I wouldn't rule out the possibility of a much nicer match existing

**Comment by engineer124** [comment]
Gave it another look and found a solution that doesn't directly access `gGameInfo->data`

**Comment by fig02** [comment]
nice

**Review by fig02 — APPROVED** [review]

**Review by Zelllll — APPROVED** [review]

**Review by Roman971 — APPROVED** [review]
